examples/texture-arrays/non-uniform.frag

```diff
@@ -0,0 +1,14 @@
+#version 450
+
+#extension GL_EXT_nonuniform_qualifier : require
+
+layout(location = 0) in vec2 v_TexCoord;
+layout(location = 1) flat in int v_Index;  // dynamically non-uniform
```

Comment:
This is actually uniform across the wave, since it's flat-interpolated. Is it supposed to be non-uniform in a difference sense?

Reply:
If you consider how gpus run fragment shaders, it may not be even coherent in the wave. They take 2x2 blocks of pixels, put N of them together into a wave. These blocks may not be from the same primative, so this may be different across the wave. This 2x2 action is also why gpus are bad at rendering small polygons, they always get a 2x2 section, even if they are only 1x1.

Reply:
My understanding is that 2x2 blocks are *always* from the same primitive. If some of these pixels lie outside, they become "ghost" threads, but regardless - `flat` is there precisely to inform the compiler that this is constant across the wave.

The link you provided is very familiar to me, although I'll make sure to rehash that part in my mind.

Reply:
The pixels from within the 2x2 blocks are always from the same primitive, being masked out and such as needed, but when 8 of those are combined into a wave, each of those 8 blocks could be from a different primitive. If that wasn't allowed, having a 1x1 triangle would waste an entire wave of processing power, not just a 2x2 block.